### PR TITLE
Death of Numbered Vermin 2.0

### PR DESCRIPTION
**Pull request by Kaedwuff**
New PR since the old one has been stagnant for weeks now and I suspect people don't notice stuff past the front page.
_________
-Removes numerical designations on every mob type I could find
-Implements a new system of randomized descriptive adjectives pulled from lists that can be later used or modified as needed. This affects rats, drones, golems, contructs, slimes, and several monkey variants. It's possible I missed things, I can add those later if needed.
-Rats can spawn randomly as vermin, mice, or rodents instead, for additional variety.
-Rats and Maint drones players can edit their name to a simple "[descriptor] [mobtype]". The game does check to make sure your chosen name is valid before allowing you to take it. Name changes generate an adminlog, so you will be noticed immediately if you start trying to be a nazi rat or some stupid shit.
-Slime players can now also change their names, with less limitations.
-A namechange helper was created for use of the former system, and can be adapted by anyone for use in a name-change proc with the addition of a regex guideline/definition.
-Adds hand labelers to to science to help with experiment designations.
-Lizards lost their typo-borne proper pronoun status and are now just 'a lizard' instead of "Lizard'.

Project thread: https://forums.aurorastation.org/topic/13125-removing-immersion-breaking-numerical-designations/

### CODE REVIEW

**Comment by Kaedwuff** [comment]
!review

**Comment by Sindorman** [comment]
Please put back numerical designations, they are so admins can tp to specific mob

**Comment by Kaedwuff** [comment]
> Please put back numerical designations, they are so admins can tp to specific mob

... This doesn't... prevent that? :I

**Comment by Sindorman** [comment]
Why are there undocumented map changes?

**Comment by Sindorman** [comment]
> > Please put back numerical designations, they are so admins can tp to specific mob
> 
> ... This doesn't... prevent that? :I
> 
> There is literally never a single time, nor will there ever be a time, when we needed unique designations for 1000 rats at once. This still gives them unique designations, it's just based on adjectives, not numbers. Read the full PR.

That is not better. List of 40 adjectives is less than 1000 unique digits. Probability of getting duplicate from 40 adjectives is so much higher than out of 1000 digits.

**Comment by Kaedwuff** [comment]
> Why are there undocumented map changes?

The hand labels were added by request of skull in the forum thread.  The changes are mentioned here now, too.

> That is not better. List of 40 adjectives is less than 1000 unique digits. Probability of getting duplicate from 40 adjectives is so much higher than out of 1000 digits. So that means you can only have at most 40(or whatever size of adjectives list is for mob type) unique name mobs of that type.

It's sufficient since people can change their mob name now and science has said hand labels. The only reason you would reasonably need to hop to a mob is because a player is inhabiting it and somehow causing problems.  There are never going to be too many players placing mice, slime, or monkies that this will cause a problem.  I even created admin logs that let you track name changes.  

You are creating an issue out of something that doesn't actually have a strong chance of coming up on our server on the off chance it somehow, maybe, happens in the future.

**Comment by NonQueueingMatt** [comment]
I don't like that rats and maint drones can change their name at all. It'll lead to some pretty dumb reoccurring shit, even if it's logged.

File: code/modules/mob/living/carbon/alien/alien.dm [hunk 1]
```diff
@@ -27,7 +30,7 @@
 	verbs += /mob/living/proc/ventcrawl
 	verbs += /mob/living/proc/hide
 
-	name = "[initial(name)] ([rand(1, 1000)])"
+	name = "([get_descriptor()] [initial(name)]"
```
Comment (LadyFowls) [inline code/modules/mob/living/carbon/alien/alien.dm]: Simple mobs, especially ghost controlled mobs should not be named in this way, its generally a un-needed bloaty change.

File: code/modules/mob/living/carbon/alien/diona/diona_nymph.dm [hunk 2]
```diff
@@ -47,6 +48,11 @@
 	var/echo = 0 //if it's an echo nymph, which has unique properties
 	var/detached = FALSE
 
+/mob/living/carbon/alien/diona/get_descriptor()
+	return pick(diona_descriptors)
+
+var/list/diona_descriptors = list("rustling","skittering","friendly","leafy","frisky","sluggish","energetic","ordinary","unremarkable","sleepy","curious","bouncy","moist","lonely","sociable","chittering","beady-eyed","silent","quiet","small","tiny","big","large","slender","thick","vibrating","naive","lost","newborn","chirping","eager","shy")
```
Comment (LadyFowls) [inline code/modules/mob/living/carbon/alien/diona/diona_nymph.dm]: This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs

File: code/modules/mob/living/carbon/human/species/station/golem.dm [hunk 3]
```diff
@@ -88,8 +88,10 @@
 	H.gender = NEUTER
 	..()
 
+var/list/golem_descriptors = list("lumbering","ponderous","slow","rumbling","sleek","solid","ephemeral","dense","shimmering","dull","glittering","shining","sluggish","quiet","ominious","lightweight","weighty","honest","watchful","short","tall","mysterious","curious","dimwitted")
+
 /datum/species/golem/get_random_name()
-	return "[golem_designation] golem ([rand(1, 1000)])"
+	return "[pick(golem_descriptors)] [golem_designation] golem"
```
Comment (LadyFowls) [inline code/modules/mob/living/carbon/human/species/station/golem.dm]: This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs

File: code/modules/mob/living/carbon/human/species/station/monkey.dm [hunk 4]
```diff
@@ -55,8 +55,10 @@
 	if(prob(1))
 		H.emote(pick("scratch","jump","roll","tail"))
 
+var/list/monkey_descriptors = list("clever","small","large","skinny","fat","tiny","noisy","playful","lazy","sleepy","energetic","listless","curious","watchful","oblivious","generic","ordinary","unremarkable","nervous","skittish","brainless",)
+
 /datum/species/monkey/get_random_name()
-	return "[lowertext(name)] ([rand(100,999)])"
+	return "[pick(monkey_descriptors)] [lowertext(name)]"
```
Comment (LadyFowls) [inline code/modules/mob/living/carbon/human/species/station/monkey.dm]: This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs

File: code/modules/mob/living/carbon/metroid/metroid.dm [hunk 5]
```diff
@@ -61,14 +61,16 @@
 	var/core_removal_stage = 0 //For removing cores.
 	var/toxloss = 0
 
+var/list/slime_descriptors = list("jiggling","wiggling","bobbing","slippery","moist","slick","translucent","transparent","opaque","sparking","buzzing","crackling","humming","dangerous","friendly","cuddly","malevolent","bouncing","harmless","rare","common","uncommon","ultra-rare","captivating","distrating","hypnotic")
```
Comment (LadyFowls) [inline code/modules/mob/living/carbon/metroid/metroid.dm]: This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 6]
```diff
@@ -146,10 +146,22 @@
 	real_name = pickedName
 	name = real_name
 
+var/regex/synthgex = regex(@"^[a-z]+(?:-[a-z]+?)? maintenance drone$") // making sure names fall within logical parameters
+
+var/list/synthetic_descriptors = list("shiny","drab","old","worn","glittering","flawed","mint-condition","generic","ordinary","unremarkable","standard","clicking","buzzing","whirring","beeping","quiet","silent","humming","small","tiny","booping")
```
Comment (LadyFowls) [inline code/modules/mob/living/silicon/robot/drone/drone.dm]: This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 7]
```diff
@@ -100,6 +100,32 @@
...
+		log_and_message_admins("changed their name to [input].", src)
+	else
+		log_and_message_admins("changed [key_name(src)]'s name to [input].", M) //To track name changes and ooc shittery.
+	name = input
+	real_name = name
+	
+
+var/regex/ratgex = regex(@"^[a-z]+(?:-[a-z]+?)? (?:rat|rodent|vermin|mouse)$") // making sure names fall within logical parameters
+
+var/list/vermin_descriptors = list("majestic","stinky","mangy","diseased","sickly","short-tailed","long-toothed","pale-furred","long-furred","long-tailed","small","large","huge","tiny","filthy","noisy","quiet","fat","skinny","emaciated","rancid","dirty")
```
Comment (LadyFowls) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]: This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs

**Comment by LadyFowls** [comment]
To address my review, you could make separate list files using the same method as I described for that txt

File: code/modules/mob/living/carbon/metroid/metroid.dm [hunk 8]
```diff
@@ -28,7 +28,7 @@
 	var/powerlevel = 0 // 0-10 controls how much electricity they are generating
 	var/amount_grown = 0 // controls how long the slime has been overfed, if 10, grows or reproduces
 
-	var/number = 0 // Used to understand when someone is talking to it
+	var/descriptor = 0 // Used to understand when someone is talking to it
```
Comment (skull132) [inline code/modules/mob/living/carbon/metroid/metroid.dm]: Use `""` or leave as null. If it's meant to be a string, then `0` is a nonsensical initial value.

File: code/modules/mob/living/carbon/metroid/metroid.dm [hunk 9]
```diff
@@ -78,6 +80,17 @@
 	coretype = text2path("/obj/item/slime_extract/[sanitizedcolour]")
 	regenerate_icons()
 
+var/regex/slimegex = regex(@"^[A-z \-]+$")
```
Comment (skull132) [inline code/modules/mob/living/carbon/metroid/metroid.dm]: This can be `var/static/...` inside the `rename_slime()` verb.

File: code/modules/mob/living/silicon/robot/drone/drone.dm [hunk 10]
```diff
@@ -146,10 +146,22 @@
 	real_name = pickedName
 	name = real_name
 
+var/regex/synthgex = regex(@"^[a-z]+(?:-[a-z]+?)? maintenance drone$") // making sure names fall within logical parameters
```
Comment (skull132) [inline code/modules/mob/living/silicon/robot/drone/drone.dm]: This can be `var/static/...` inside the `rename_drone` verb.

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 11]
```diff
@@ -100,6 +100,32 @@
...
+
+	if (M == src)
+		log_and_message_admins("changed their name to [input].", src)
+	else
+		log_and_message_admins("changed [key_name(src)]'s name to [input].", M) //To track name changes and ooc shittery.
+	name = input
+	real_name = name
+	
+
+var/regex/ratgex = regex(@"^[a-z]+(?:-[a-z]+?)? (?:rat|rodent|vermin|mouse)$") // making sure names fall within logical parameters
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]: This can be static and not a global.

File: code/modules/mob/living/carbon/human/species/station/monkey.dm [hunk 12]
```diff
@@ -55,8 +55,10 @@
 	if(prob(1))
 		H.emote(pick("scratch","jump","roll","tail"))
 
+var/list/monkey_descriptors = list("clever","small","large","skinny","fat","tiny","noisy","playful","lazy","sleepy","energetic","listless","curious","watchful","oblivious","generic","ordinary","unremarkable","nervous","skittish","brainless",)
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/species/station/monkey.dm]: Leftover comma before the ) at the end of this list

File: code/modules/mob/living/carbon/alien/diona/diona_nymph.dm (hunk 2 above)
Comment (Kaedwuff) [reply code/modules/mob/living/carbon/alien/diona/diona_nymph.dm]: This is more copypasted conjecture than review, I'm.. not sure you actually read the adjectives file you directed me to, because using it would result in nonsense like:

-dead rat
-misty nymph
-impossible monkey
-rich mouse
-wrong golem

By contrast, my lists are catered to the specific mobs they are meant for (diseased rat, ponderous golem, etc), and I find it somewhat insulting that you consider the effort to make content tailored what it is being used for 'bloaty' and would prefer I use a text file that contains about 25% object and location based adjectives in it.

File: code/modules/mob/living/carbon/alien/alien.dm (hunk 1 above)
Comment (Kaedwuff) [reply code/modules/mob/living/carbon/alien/alien.dm]: Why is it not needed. Can you explain why it is bloaty? This is a fundamental component of the PR, and you have not provided any context as to why it needs to be changed other than 'I think it's unncessary'

**Comment by Kaedwuff** [comment]
> To address my review, you could make separate list files using the same method as I described for that txt

That actually DOES seem bloaty. Why should I make half a dozen list files when I can just write a list in the code?

**Comment by NonQueueingMatt** [comment]
Massive amount of merge conflicts + unaddressed reviews. Reopen when ready.
